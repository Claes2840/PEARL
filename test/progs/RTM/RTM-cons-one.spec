// Inner variables
Q = 'nil
Q1 = 'nil
S1 = 'nil
S2 = 'nil
Q2 = 'nil
RulesRev = 'nil
Rules = 'nil
Rule = 'nil

// Describe the TM
// Cons 1 to any input
// Example: _101 --> _1101
Start = '1
End = '4
pc_max = '3
Transitions = 
 '((1 . (BLANK . (1     . 2))) . 
  ((2 . (SLASH . (LEFT  . 3))) . 
  ((3 . (BLANK . (BLANK . 4))) . 
   nil))))))))

// Tape for full specialization
//S = 'nil
//S_left = 'nil
//S_right = '(1 . (0 . (1 . nil)))
