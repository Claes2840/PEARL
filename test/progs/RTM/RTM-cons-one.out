(S_left S S_right) -> (S_left S S_right)

init_2:
	entry
	assert((S = 'BLANK))
	S ^= 'BLANK
	S ^= '1
	assert((S = '1))
	assert(!((S = 'SLASH)))
	if ((S_right = 'nil) && (S = 'BLANK))
		goto left_1b_1
		else left_1p_1

left_1b_1:
	from init_2
	S ^= 'BLANK
	goto left1_1

left1_1:
	fi (S_right = 'nil)
		from left_1b_1
		else left_1p_1
	if (S_left = 'nil)
		goto left_2b_1
		else left_2p_1

left_2b_1:
	from left1_1
	S ^= 'BLANK
	goto left2_1

left2_1:
	fi ((S_left = 'nil) && (S = 'BLANK))
		from left_2b_1
		else left_2p_1
	assert((S = 'BLANK))
	S ^= 'BLANK
	S ^= 'BLANK
	assert((S = 'BLANK))
	exit

left_2p_1:
	from left1_1
	(S . S_left) <- S_left
	goto left2_1

left_1p_1:
	from init_2
	S_right <- (S . S_right)
	goto left1_1
